packages/playwright/src/index.ts

```diff
@@ -678,11 +680,7 @@ class ArtifactsRecorder {
   }
 
   private async _screenshotOnTestFailure() {
-    const contexts: BrowserContext[] = [];
-    for (const browserType of [this._playwright.chromium, this._playwright.firefox, this._playwright.webkit])
-      contexts.push(...(browserType as any)._contexts);
-    const pages = contexts.map(ctx => ctx.pages()).flat();
-    await Promise.all(pages.map(page => this._screenshotPage(page, false)));
+    await Promise.all(this._playwright._allPages().map(page => this._screenshotPage(page, false)));
```

Comment:
What about places like this one: https://github.com/microsoft/playwright/blob/main/packages/playwright/src/index.ts#L625?